keras_cv/models/backbones/resnet_v2/resnet_v2_backbone.py

```diff
@@ -482,189 +388,240 @@ def get_config(self):
+            when `include_top` is `False`.
+            - `None` means that the output of the model will be
+                the 4D tensor output of the
+                last convolutional layer.
+            - `avg` means that global average pooling
+                will be applied to the output of the
+                last convolutional layer, and thus
+                the output of the model will be a 2D tensor.
+            - `max` means that global max pooling will
+                be applied.
+        weights: one of `None` (random initialization), a pretrained weight file
+            path, or name of pretrained checkpoint if available.
+            Available checkpoints: "{aliases}"
+"""
+
+
+def _backbone_alias(
+    preset,
+    include_rescaling=True,
+    input_shape=(None, None, 3),
+    input_tensor=None,
+    pooling=None,
     weights=None,
+    **kwargs,
+):
+    """Helper method to generate keras.application style interfaces for presets."""
+    # Pack args in kwargs
+    kwargs.update(
+        {
+            "include_rescaling": include_rescaling,
+            "input_shape": input_shape,
+            "input_tensor": input_tensor,
+            "pooling": pooling,
+        }
+    )
+    # Check if `weights` is valid preset alias
+    aliases = applications_weight_aliases[preset]
+    if weights and weights in aliases:
+        preset = applications_presets[(preset, weights)]
+        return ResNetV2Backbone.from_preset(preset, **kwargs)
+    # Otherwise check if valid file
+    elif weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either `None`, the path to the "
+            'weights file to be loaded, or one of valid checkpoint types: "{aliases}". '
+            "Weights file not found at location: {weights}.".format(
+                aliases=", ".join(aliases),
+                weights=weights,
+            )
+        )
+
+    model = ResNetV2Backbone.from_preset(preset, **kwargs)
+
+    if weights:
+        model.load_weights(weights)
+
+    return model
+
+
+def ResNet18V2Backbone(
+    include_rescaling=True,
     input_shape=(None, None, 3),
     input_tensor=None,
     pooling=None,
-    classifier_activation="softmax",
-    name="resnet18",
+    weights=None,
     **kwargs,
 ):
-    """Instantiates the ResNet18 architecture."""
-
-    return ResNetV2(
-        stackwise_filters=MODEL_CONFIGS["ResNet18V2"]["stackwise_filters"],
-        stackwise_blocks=MODEL_CONFIGS["ResNet18V2"]["stackwise_blocks"],
-        stackwise_strides=MODEL_CONFIGS["ResNet18V2"]["stackwise_strides"],
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        name=name,
-        weights=weights,
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="basic_block",
+    return _backbone_alias(
```

Comment:
just a comment:

now that we don't have to specify include_rescaling or include_top it is REALLY cool that you can write

ResNet18V2() and have a model!